src/routes/saved-report.tsx

```diff
@@ -1,60 +1,66 @@
 import { getSavedReports } from '@/api/report/submit-report'
-import { ReportPage } from './report'
 import Page from '@/components/page'
 import { css } from '@linaria/core'
-import { route, createAlert } from '@/router'
 import { useState, useEffect } from 'preact/hooks'
 import { Report } from '@/api/report'
-import { AlertType } from '@/components/alert'
+import Card from '@/components/card'
+import { ReportViewer } from '@/components/report-viewer'
+import { useJWT } from '@/jwt'
 
 const missingReportStyle = css`
   padding: 2rem;
   text-align: center;
 `
 
+const reportPageStyle = css`
+  display: flex;
+  padding: 2rem;
+  justify-content: center;
+`
+
+const reportViewerCardStyle = css`
+  padding: 2rem;
+  display: grid;
+  justify-items: center;
+  grid-gap: 1rem;
+  max-width: 30rem;
+`
+
+const ReportPage = ({ report }: { report: Report }) => {
+  const { jwt } = useJWT()
+  const canEdit =
+    jwt &&
+    (report.reporterId === Number.parseInt(jwt.sub) ||
+      (jwt.peregrineRoles.isAdmin && report.realmId === jwt.peregrineRealm) ||
+      jwt.peregrineRoles.isSuperAdmin)
+  return (
+    <Page name="Report" class={reportPageStyle}>
+      {/* shows the report */}
+      <Card class={reportViewerCardStyle}>
+        <ReportViewer
+          report={report}
+          reportEditorLink={
+            canEdit ? `/saved-reports/${report.key}/edit` : undefined
+          }
+        />
+      </Card>
+    </Page>
+  )
+}
```

Comment:
Before this PR, `ReportPage` was shared between the offline report editor and the "online" report editor. It looks like you split it out here. My concern is that the two pages (which are intended to look and work essentially the same) may now get out of sync in case someone forgets to edit both. For example, someone may make a CSS change in one but not the other. The only real difference between the two is how they handle fetching/storing the report data. Is there a way to combine them back together?

Reply:
I did what I thought would work well here, but I didn't do a thorough comparison of every pair of files among the six that existed. There isn't enough overlap between the viewer and the editor to consolidate.